Sources/Extensions/NSDictionary+Toggles.swift

```diff
@@ -0,0 +1,39 @@
+//  NSDictionary+Toggles.swift
+
+import Foundation
+
+extension NSDictionary {
+    public enum ToggleConversionError: Error {
+        case nsDictionaryInvalidFormat
+        case nsDictionaryEmpty
+    }
+    
+    public func convertToToggleDataSource() throws -> [Variable: Toggles.Value] {
+        guard let swiftDictionary = self as? [String: Any], !swiftDictionary.isEmpty else {
+            throw ToggleConversionError.nsDictionaryEmpty
+        }
+        
+        var resultDictionary: [Variable: Toggles.Value] = [:]
```

Comment:
We can improve this implementation with some functional programming by using `mapValues`:

```
try mapValues { value in
    switch value {
    case let stringValue as String:
        return Toggles.Value.string(stringValue)
    case let boolValue as Bool:
        return Toggles.Value.bool(boolValue)
    case let intValue as Int:
        return Toggles.Value.int(intValue)
    case let numberValue as Double:
        return Toggles.Value.number(numberValue)
    default:
        let valueType = String(describing: type(of: value))
        throw ToggleConversionError.unsupportedValueType(valueType)
    }
}
```